### PR TITLE
Fix for incorrect locking in GetPubKey() (keystore.cpp)

### PR DESCRIPTION
**Pull request by rodentrabies**
Fix for issue #7901

### CODE REVIEW

File: src/keystore.cpp [hunk 1]
```diff
@@ -19,6 +19,7 @@ bool CBasicKeyStore::GetPubKey(const CKeyID &address, CPubKey &vchPubKeyOut) con
 {
     CKey key;
     if (!GetKey(address, key)) {
+	LOCK(cs_KeyStore);
```
Comment (sipa) [inline src/keystore.cpp]: Nit: indendation error

**Comment by sipa** [comment]
utACK, but fix the indendation

**Comment by yurizhykin** [comment]
Oh, yes, sorry. I thought I have no-tabs mode for C++ in Emacs.

**Comment by MarcoFalke** [comment]
@yurizhykin Mind to squash the commits?

**Comment by jonasschnelli** [comment]
utACK (but please squash commits)

**Comment by yurizhykin** [comment]
Yep, squashed.

**Comment by paveljanik** [comment]
ACK https://github.com/bitcoin/bitcoin/pull/7913/commits/220f950ab1302f8f8c0bb5b9c818f3d6569796bd

**Comment by jonasschnelli** [comment]
ACK 220f950ab1302f8f8c0bb5b9c818f3d6569796bd

**Comment by dcousens** [comment]
utACK 220f950

**Pull request merged**